kitsune/sumo/utils.py

```diff
@@ -313,3 +313,22 @@ def get_browser(user_agent):
     else:
         browser = None
     return browser
+
+
+def is_toll_free_number(data):
+    """Figure out if we have a toll free number."""
+    digits = filter(type(data).isdigit, data)
+    if not digits:
+        return False
+
+    if settings.TOLL_FREE_REGEX.match(digits):
+        return True
+    return False
+
+@timeout(seconds=settings.REGEX_TIMEOUT)
+def match_regex_with_timeout(compiled_regex, data):
+    """Matches the specified regex.
+
+    Adds a timeout to avoid catastrophic backtracking.
+    """
+    return any(compiled_regex.findall(data))
```

Comment:
I think `compiled_regex.search(data)` will be marginally faster since it'll return on the first match

Reply:
I guess we'll need findall anyway for the allowlist, if we implement that